modules/gapi/samples/onevpl_infer_single_roi.cpp

```diff
@@ -148,15 +152,25 @@ GAPI_OCV_KERNEL(OCVLocateROI, LocateROI) {
     static void run(const cv::Size& in_size, cv::Rect &out_rect) {
 
         // Identify the central point & square size (- some padding)
-        const auto center = cv::Point{in_size.width/2, in_size.height/2};
-        auto sqside = std::min(in_size.width, in_size.height);
-
-        // Now build the central square ROI
-        out_rect = cv::Rect{ center.x - sqside/2
-                           , center.y - sqside/2
-                           , sqside
-                           , sqside
-                           };
+        // NB: GPU plugin in InferenceEngine doesn't support ROI at now
+        if (device_id.find("GPU") == std::string::npos) {
```

Comment:
From my perspective `flag` is redundant here because it duplicates entity device and all information about `flag` can be extracted from `device_id`.  + duplicated entity would become inconsistent when the main entity `device_id` changed for some reason

If you don't like huge check on device type then i propose to introduce global function `bool is_gpu(string)` instead of flag

Reply:
Yes, these unnesessary repetitions -  I just suggested to reduce their amount.  
The function's call is a repetition too, but it's OK for me - yes, I think it is an OK solution :)